File: Ruby/test/test_signature.rb
require 'test/unit'
require 'json'
require './signature'

class TestSignature < Test::Unit::TestCase
  def test_signature
    sample = JSON.load(File.read('../sample.txt'))
    tool = MeiqiaSignTools.new(sample['key'])
    sample['maps'].each do |map|
      assert_equal(map['sign'], tool.sign(map['text']))
    end
  end
end

File: Python/tests/test_signature.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-import unittest
import unittest
import json
from signature import MTSigner

class TestSignature(unittest.TestCase):
    def setUp(self):
        pass
            
    def tearDown(self):
        pass

    def testSign(self):
    	fo = open("../sample.txt", "r")
    	str = fo.read()
    	fo.close()
    	sample = json.loads(str)
    	
    	signer = MTSigner(sample['key'].encode('utf-8'))
    	for _map in sample['maps']:
    		self.assertEqual(_map['sign'], signer.sign(_map['text'].encode('utf-8')))
